File: app/buildz/tmp/kapt3/stubs/debug/com/bits/bee/bpmc/data/repository/PmtdRepositoryImpl.java
package com.bits.bee.bpmc.data.repository;

import java.lang.System;

/**
 * Created by aldi on 30/03/22.
 */
@kotlin.Metadata(mv = {1, 7, 1}, k = 1, d1 = {"\u0000D\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\u0010 \n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\u0018\u00002\u00020\u0001B7\b\u0007\u0012\u0006\u0010\u0002\u001a\u00020\u0003\u0012\u0006\u0010\u0004\u001a\u00020\u0005\u0012\u0006\u0010\u0006\u001a\u00020\u0007\u0012\u0006\u0010\b\u001a\u00020\t\u0012\u0006\u0010\n\u001a\u00020\u000b\u0012\u0006\u0010\f\u001a\u00020\r\u00a2\u0006\u0002\u0010\u000eJ\u0014\u0010\u000f\u001a\u000e\u0012\n\u0012\b\u0012\u0004\u0012\u00020\u00120\u00110\u0010H\u0016J\u001a\u0010\u0013\u001a\u0014\u0012\u0010\u0012\u000e\u0012\n\u0012\b\u0012\u0004\u0012\u00020\u00120\u00110\u00140\u0010H\u0016R\u000e\u0010\u0002\u001a\u00020\u0003X\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u000e\u0010\n\u001a\u00020\u000bX\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u000e\u0010\f\u001a\u00020\rX\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u000e\u0010\u0006\u001a\u00020\u0007X\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u000e\u0010\b\u001a\u00020\tX\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u000e\u0010\u0004\u001a\u00020\u0005X\u0082\u0004\u00a2\u0006\u0002\n\u0000\u00a8\u0006\u0015"}, d2 = {"Lcom/bits/bee/bpmc/data/repository/PmtdRepositoryImpl;", "Lcom/bits/bee/bpmc/domain/repository/PmtdRepository;", "apiUtils", "Lcom/bits/bee/bpmc/data/data_source/remote/ApiUtils;", "pmtdDao", "Lcom/bits/bee/bpmc/data/data_source/local/dao/PmtdDao;", "edcDao", "Lcom/bits/bee/bpmc/data/data_source/local/dao/EdcDao;", "edcSurcDao", "Lcom/bits/bee/bpmc/data/data_source/local/dao/EdcSurcDao;", "branchDao", "Lcom/bits/bee/bpmc/data/data_source/local/dao/BranchDao;", "defaultDispatcher", "Lkotlinx/coroutines/CoroutineDispatcher;", "(Lcom/bits/bee/bpmc/data/data_source/remote/ApiUtils;Lcom/bits/bee/bpmc/data/data_source/local/dao/PmtdDao;Lcom/bits/bee/bpmc/data/data_source/local/dao/EdcDao;Lcom/bits/bee/bpmc/data/data_source/local/dao/EdcSurcDao;Lcom/bits/bee/bpmc/data/data_source/local/dao/BranchDao;Lkotlinx/coroutines/CoroutineDispatcher;)V", "getActivePmtdList", "Lkotlinx/coroutines/flow/Flow;", "", "Lcom/bits/bee/bpmc/domain/model/Pmtd;", "getLatestPmtdList", "Lcom/bits/bee/bpmc/utils/Resource;", "app_debug"})
public final class PmtdRepositoryImpl implements com.bits.bee.bpmc.domain.repository.PmtdRepository {
    private final com.bits.bee.bpmc.data.data_source.remote.ApiUtils apiUtils = null;
    private final com.bits.bee.bpmc.data.data_source.local.dao.PmtdDao pmtdDao = null;
    private final com.bits.bee.bpmc.data.data_source.local.dao.EdcDao edcDao = null;
    private final com.bits.bee.bpmc.data.data_source.local.dao.EdcSurcDao edcSurcDao = null;
    private final com.bits.bee.bpmc.data.data_source.local.dao.BranchDao branchDao = null;
    private final kotlinx.coroutines.CoroutineDispatcher defaultDispatcher = null;
    
    @javax.inject.Inject()
    public PmtdRepositoryImpl(@org.jetbrains.annotations.NotNull()
    com.bits.bee.bpmc.data.data_source.remote.ApiUtils apiUtils, @org.jetbrains.annotations.NotNull()
    com.bits.bee.bpmc.data.data_source.local.dao.PmtdDao pmtdDao, @org.jetbrains.annotations.NotNull()
    com.bits.bee.bpmc.data.data_source.local.dao.EdcDao edcDao, @org.jetbrains.annotations.NotNull()
    com.bits.bee.bpmc.data.data_source.local.dao.EdcSurcDao edcSurcDao, @org.jetbrains.annotations.NotNull()
    com.bits.bee.bpmc.data.data_source.local.dao.BranchDao branchDao, @org.jetbrains.annotations.NotNull()
    kotlinx.coroutines.CoroutineDispatcher defaultDispatcher) {
        super();
    }
    
    @org.jetbrains.annotations.NotNull()
    @java.lang.Override()
    public kotlinx.coroutines.flow.Flow<com.bits.bee.bpmc.utils.Resource<java.util.List<com.bits.bee.bpmc.domain.model.Pmtd>>> getLatestPmtdList() {
        return null;
    }
    
    @org.jetbrains.annotations.NotNull()
    @java.lang.Override()
    public kotlinx.coroutines.flow.Flow<java.util.List<com.bits.bee.bpmc.domain.model.Pmtd>> getActivePmtdList() {
        return null;
    }
}

File: app/src/main/java/com/bits/bee/bpmc/data/repository/PmtdRepositoryImpl.kt
package com.bits.bee.bpmc.data.repository

import com.bits.bee.bpmc.data.data_source.local.dao.BranchDao
import com.bits.bee.bpmc.data.data_source.local.dao.EdcDao
import com.bits.bee.bpmc.data.data_source.local.dao.EdcSurcDao
import com.bits.bee.bpmc.data.data_source.local.dao.PmtdDao
import com.bits.bee.bpmc.data.data_source.local.model.PmtdEntity
import com.bits.bee.bpmc.data.data_source.remote.ApiUtils
import com.bits.bee.bpmc.domain.mapper.PmtdDataMapper
import com.bits.bee.bpmc.domain.model.Pmtd
import com.bits.bee.bpmc.domain.repository.PmtdRepository
import com.bits.bee.bpmc.utils.Resource
import kotlinx.coroutines.CoroutineDispatcher
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.flow
import kotlinx.coroutines.flow.flowOn
import javax.inject.Inject

/**
 * Created by aldi on 30/03/22.
 */
class PmtdRepositoryImpl @Inject constructor(
    private val apiUtils: ApiUtils,
    private val pmtdDao: PmtdDao,
    private val edcDao: EdcDao,
    private val edcSurcDao: EdcSurcDao,
    private val branchDao : BranchDao,
    private val defaultDispatcher: CoroutineDispatcher
) : PmtdRepository {

    override fun getLatestPmtdList(): Flow<Resource<List<Pmtd>>> {
//        return object : NetworkDatabaseBoundResource<List<PmtdEntity>, PmtdResponse>(){
//            override suspend fun loadFormDB(): List<PmtdEntity>? {
//                return pmtdDao.getPmtdList()
//            }
//
//            override fun shouldFetch(data: List<PmtdEntity>?): Boolean {
//                return true
//            }
//
//            override suspend fun createCall(): Flow<ApiResponse<PmtdResponse>> {
//                return apiUtils.getPmtdApiService().getPmtdList()
//            }
//
//            override suspend fun saveCallResult(data: PmtdResponse) {
//                pmtdDao.insertBulk(data.data.data.map { it.toPmtd() })
//            }
//        }.getAsFlow()
        return flow {
            val edcList = edcDao.getEdcList()
            var pmtNewList = mutableListOf<PmtdEntity>()

            for (edc in edcList){
                val edcSurcList = edcSurcDao.getEdcSurcByEdc(edc.id)
                val branch = edc.branchId?.let { branchDao.getBranchById(it) }
                for (edcSurc in edcSurcList){
                    val pmtd = PmtdEntity(
                        branchId = edc.branchId,
                        branchName = branch?.name,
                        surAcc = edcSurc.surcAccId,
                        surExp = edcSurc.surcExp,
                        cashId = edc.cash,
                        ccType = edcSurc.ccType,
                        mdrAcc = edcSurc.mdrAccId,
                        mdrExp = edcSurc.mdrExp,
                        code = edc.code,
                        name = edc.name,
                        cashName = null,
                        id = null,
                        settleDays = null,
                        idx = null,
                        edcSurcType = edcSurc.edcSurcType
                    )

                    pmtNewList.add(pmtd)
                }
            }

            pmtdDao.insertBulk(pmtNewList)

            val pmtdList = pmtdDao.getPmtdList()
            emit(Resource.success(pmtdList.map { PmtdDataMapper.fromDbToDomain(it) }))
        }.flowOn(defaultDispatcher)
    }

    override fun getActivePmtdList(): Flow<List<Pmtd>> = flow<List<Pmtd>> {
        val data = pmtdDao.getPmtdList()
        emit( data.map { PmtdDataMapper.fromDbToDomain(it) })
    }.flowOn(defaultDispatcher)
}